sgx/untrusted/src/ocall/ecdh_ocall.c

```diff
@@ -15,33 +50,142 @@ int ecdh_exchange_ocall(unsigned char *enclave_ephemeral_public,
                         unsigned char *enclave_eph_pub_signature,
                         unsigned int enclave_eph_pub_signature_len,
                         unsigned char **remote_ephemeral_public,
-                        size_t * remote_ephemeral_public_len,
+                        size_t *remote_ephemeral_public_len,
                         unsigned char **remote_eph_pub_signature,
-                        unsigned int *remote_eph_pub_signature_len)
+                        unsigned int *remote_eph_pub_signature_len,
+                        int socket_fd)
+{
+  int ret;
+
+  kmyth_log(LOG_DEBUG, "Sending ephemeral public key.");
+  ret =
+    write(socket_fd, &enclave_ephemeral_public_len,
+          sizeof(enclave_ephemeral_public_len));
+  if (ret == -1)
```

Comment:
This is probably kind of a trivial comment, but is it more robust to check if ret != 0 instead of checking that ret == -1. I actually am not an expert on the socket API, so I am good with whatever is considered best practice. Note: this applies to a number of conditionals in this file, but I will only mention it here.

Reply:
Yes, it's generally better to compare the return value to the expected read/write length.
This catches under-reads and under-writes (which aren't really a concern here)
and also detects when we read from a socket that just closed.